### PR TITLE
[Xedra Evolved] Add Time Related Monsters

### PR DESCRIPTION
**Pull request by b3brodie**
#### Summary
Mods "[Xedra Evolved] Add Time Related Monsters"

#### Purpose of change
Sub PR of #78366
Adds a couple of time related enemies for for time content and to also add enemies for chronomancers to kill to gain insight.

#### Describe the solution
Adds 3 time related zombies
- Monochrome Zombie: Equal stats to baseline zombie, but resets health every 10 seconds.
- Destabilized Zombie: Equal stats to baseline zombie other than 1.5x speed and health. Also, it is immune to time freezes and can travel through frozen time. Finally, its health resets every 5 seconds.
- Zombie Timewound: Adds zombie timewound. Equal stats to baseline zombie other than 2x speed and health. In addition to time freeze immunity, its attacks apply a dot if they connect. Finally, its health resets every 2 seconds.

Distributes STABILIZED_TIMELINE flag to a couple of other monsters, such as hounds of tindalos.

Infrastructure:
- Allows monsters to have hit_you and hit_me enchantment effects
- Allows monsters to move through impassable fields if they have the immunity data
- Adds has_flag method for monsters that checks for flag_ids rather than the prior specific monster_flag_id, which allows for more dynamic checking.

#### Describe alternatives you've considered
One big PR

#### Testing
Tested that zombies correctly restore their HP
Tested that monsters with the STABILIZED_TIMELINE flag correctly path through fields they are immune to
Tested that monsters apply the spell for their hit_you enchants
Tested that monsters apply the spell for their hit_me enchants
Teleported around a bit and confirmed that the monsters were actually appearing

### CODE REVIEW

File: data/mods/Xedra_Evolved/monsters/zombies.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        "cooldown": 2,
+        "monster_message": "%1$s flickers and all of its injuries disappear!"
+      },
+      {
+        "id": "xedra_monster_erosion_buff_special",
+        "type": "spell",
+        "condition": { "not": { "u_has_effect": "effect_xedra_monster_erosion" } },
+        "spell_data": { "id": "xedra_monster_erosion_buff" },
+        "allow_no_target": true,
+        "monster_message": "the ground around %1$s starts eroding into nothingness."
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/monsters/zombies.json:77]:
```suggestion
        "monster_message": "The ground around %1$s starts eroding into nothingness."
```

**Comment by Standing-Storm** [comment]
> Allows monsters to have hit_you and hit_me enchantment effects

Oh, the hell which I shall now unleash

File: src/magic_enchantment.cpp [hunk 2]
```diff
@@ -352,6 +352,11 @@ bool enchantment::is_monster_relevant() const
         }
     }
 
+    // check for hit you / me effects
+    if( hit_you_effect.size() > 0 || hit_me_effect.size() > 0 ) {
```
Comment (Maleclypse) [inline src/magic_enchantment.cpp:356]: Clang errors?

Comment (b3brodie) [reply src/magic_enchantment.cpp]: Huh, I missed that for the other seemingly? unrelated errors.  Switched over to empty()

**Comment by Maleclypse** [comment]
Weird is that clang failure related to this PR or something in master?

**Comment by b3brodie** [comment]
> Weird is that clang failure related to this PR or something in master?

I dont believe that is this pr.  Currently #78565 has the same clang failures.

**Pull request merged**